framework/src/interfacekernels/ADInterfaceKernel.C

```diff
@@ -56,7 +56,7 @@ ADInterfaceKernelTempl<T>::ADInterfaceKernelTempl(const InputParameters & parame
     _grad_neighbor_value(_neighbor_var.adGradSlnNeighbor()),
     _phi_neighbor(_assembly.phiFaceNeighbor(_neighbor_var)),
     _test_neighbor(_neighbor_var.phiFaceNeighbor()),
-    _grad_test_neighbor(_neighbor_var.adGradPhiFaceNeighbor())
+    _grad_test_neighbor(_neighbor_var.gradPhiFaceNeighbor())
 
 {
```

Comment:
Is it worth printing a warning message if the user sets `use_displaced_mesh = true`? I think a user would expect to get exact Jacobian by using AD, but they won't if the weak form involves grad_test or grad_phi.

Reply:
It's really only wrong if someone uses `_grad_test_neighbor` which I can't really detect. For most people this would be a false positive. I could stop providing this in the base class and then if a derived class requests it then I could give a warning

Reply:
That's true. How long do you think it will take until someone implements AD neighbor assembly? If it doesn't take long, then maybe a comment is enough. A warning can prevent us from forgetting about this issue -- every time a user runs the AD interface kernel on the displaced mesh _and_ requests grad test/phi, tell them to find Alex and complain :J

Reply:
:laughing:

I think it will be a while until we do AD neighbor assembly in MOOSE. We are exploring tackling this all at the libMesh level in https://github.com/libMesh/libmesh/pull/2576 but it is slow progress as other demands come up and there doesn't seem to be a huge customer. I know that both @recuero and I would like to test it out on mortar but I am a little skeptical that the added cost will outweigh any performance improvement over using a "pretty good Jacobian" with PJFNK (or even NEWTON).

My thoughts: we have the doxygen comment for `_grad_test_neighbor` that is not using AD. I will also create an issue. I'm reticent to do much more as non-important warnings tend to lead to users ignoring important warnings. And currently there are no users (to my knowledge) of a `_grad_test_neighbor` on a displaced mesh with an AD object

Reply:
I've opened https://github.com/idaholab/moose/issues/24480

Reply:
👍 👍